felix/bpf/libbpf/libbpf_api.h

```diff
@@ -481,3 +485,70 @@ int num_possible_cpu()
 {
     return libbpf_num_possible_cpus();
 }
+
+int create_bpf_map(enum bpf_map_type type, unsigned int key_size, unsigned int value_size,
+                   unsigned int max_entries, unsigned int flags, const char *name) {
+	LIBBPF_OPTS(bpf_map_create_opts, create_attr);
+
+	create_attr.map_flags = flags;
+
+	switch (type) {
+	case BPF_MAP_TYPE_PERF_EVENT_ARRAY:
+	case BPF_MAP_TYPE_CGROUP_ARRAY:
+	case BPF_MAP_TYPE_STACK_TRACE:
+	case BPF_MAP_TYPE_ARRAY_OF_MAPS:
+	case BPF_MAP_TYPE_HASH_OF_MAPS:
+	case BPF_MAP_TYPE_DEVMAP:
+	case BPF_MAP_TYPE_DEVMAP_HASH:
+	case BPF_MAP_TYPE_CPUMAP:
+	case BPF_MAP_TYPE_XSKMAP:
+	case BPF_MAP_TYPE_SOCKMAP:
+	case BPF_MAP_TYPE_SOCKHASH:
+	case BPF_MAP_TYPE_QUEUE:
+	case BPF_MAP_TYPE_STACK:
+	case BPF_MAP_TYPE_ARENA:
+		create_attr.btf_fd = 0;
+		create_attr.btf_key_type_id = 0;
+		create_attr.btf_value_type_id = 0;
+		break;
+	case BPF_MAP_TYPE_STRUCT_OPS:
+		create_attr.btf_value_type_id = 0;
+		break;
+	default:
+		break;
+	}
```

Comment:
Afaict this only sets to zero what should be already zeroed https://github.com/libbpf/libbpf/blob/fc687b8ee934b7331fa6561311346ce8cf8086fa/src/libbpf_common.h#L60-L64